project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,456 @@
+/*
+    © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define("project_cost_report", function (require) {
+"use strict";
+
+var ControlPanelMixin = require("web.ControlPanelMixin");
+var core = require("web.core");
+var crashManager = require("web.crash_manager");
+var framework = require("web.framework");
+var rpc = require("web.rpc");
+var session = require("web.session");
+var Widget = require("web.Widget");
+
+var QWeb = core.qweb;
+var _t = core._t;
+
+/**
+ * Get a record ID from an event triggered.
+ *
+ * The record ID is expected to be stored in a data-id attribute on the event target.
+ *
+ * @param {jQuery.Event} - the triggered event
+ * @returns {Number} - the record ID
+ */
+function getRecordIdFromEvent(event){
+    var attributeNode = event.currentTarget.attributes["data-id"];
+    return attributeNode ? parseInt(attributeNode.nodeValue) : false;
+}
+
+/**
+ * Get a record name from an event triggered.
+ *
+ * The record name is expected to be stored in a data-name attribute on the event target.
+ *
+ * @param {jQuery.Event} - the triggered event
+ * @returns {String} - the record name
+ */
+function getRecordNameFromEvent(event){
+    var attributeNode = event.currentTarget.attributes["data-name"];
+    return attributeNode ? attributeNode.nodeValue : false;
+}
+
+var ReportAction = Widget.extend(ControlPanelMixin, {
+    events: {
+        "click .o_project_cost_report__analytic_line": "analyticLineClicked",
+        "click .o_project_cost_report__product_category": "productCategoryClicked",
+        "click .o_project_cost_report__product_total": "productTotalClicked",
+        "click .o_project_cost_report__time_category": "timeCategoryClicked",
+        "click .o_project_cost_report__time_total_hours": "hourTotalClicked",
+        "click .o_project_cost_report__time_total": "hourTotalClicked",
+        "click .o_project_cost_report__outsourcing_category": "outsourcingCategoryClicked",
+        "click .o_project_cost_report__outsourcing_total": "outsourcingTotalClicked",
+        "click .o_project_cost_report__purchase_order_name": "purchaseOrderNameClicked",
+    },
+    getCategoryNames(){
+        return ["product", "time", "outsourcing"];
+    },
+    init(parent, action) {
+        this._super.apply(this, arguments);
+        this.controllerURL = action.context.url;
+        this.projectId = action.context.active_id || action.params.active_id;
+        this.reportContext = {
+            active_id: this.projectId,
+            unfolded_categories: {},
+            show_summary: true,
```

Comment:
Codacy found an issue: [Identifier 'show_summary' is not in camel case.](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6119862)